### DISCUSSION TITLE
关于killthread

### DISCUSSION THREAD

**Discussion by mippo520** [cloudwu/skynet#1789] [General]
A服务有多条send和call到B服务,其中send的一条卡死了,我使用killthread想关闭这个卡死的协程,结果是不仅关闭了卡死的这个协程,还把A服务中等待B服务返回的call的协程全部都关掉了.
我看了下代码,send的session为0,killthread中会发送PTYPE_ERROR,如果session为0了那么A服务会把所有等待B服务返回的都关掉.我想问下这是为什么,session=0的时候就把这个协程关掉然后不发回去不行吗

**Comment by cloudwu** [comment]
最新的提交增加了 session > 0 检查。
